### PR TITLE
Dead Code

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
Removed dead code.

Why should this be merged: Dead code. 

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: common/src/main/java/org/corfudb/common/result/Result.java [hunk 1]
```diff
@@ -126,13 +126,4 @@ public void ifError(Consumer<E> consumer) {
             return new Result<>(null, error);
         }
     }
-
-    public<U> Result<U, E> flatMap(Function<T, Result<U, E>> mapper) {
```
Comment (xnull) [inline common/src/main/java/org/corfudb/common/result/Result.java]: this code is requred

Comment (Maithem) [reply common/src/main/java/org/corfudb/common/result/Result.java]: where is it being used?

From `Supplier.java`

```java

@FunctionalInterface
public interface Supplier<T> {

    /**
     * Gets a result.
     *
     * @return a result
     */
    T get();
} ```

Comment (xnull) [reply common/src/main/java/org/corfudb/common/result/Result.java]: It's not used yet, but it's a part of `Result` API and should not be deleted

Comment (Maithem) [reply common/src/main/java/org/corfudb/common/result/Result.java]: ok

Comment (Maithem) [reply common/src/main/java/org/corfudb/common/result/Result.java]: Done.

**Pull request merged**